io/EptReader.cpp

```diff
@@ -290,83 +283,76 @@ QuickInfo EptReader::inspect()
 
 void EptReader::addDimensions(PointLayoutPtr layout)
 {
-    const Json::Value& schema(m_info->schema());
+    const NL::json schema {m_info->schema()};
     m_remoteLayout.reset(new FixedPointLayout());
 
-    for (Json::ArrayIndex i(0); i < schema.size(); ++i)
+    for (auto& el : schema)
     {
-        const Json::Value& dim(schema[i]);
-        const std::string name(dim["name"].asString());
+        const std::string name(el["name"].get<std::string>());
```

Comment:
https://github.com/PDAL/PDAL/issues/2495

For me, this line causes 
```
unknown file: Failure
C++ exception with description "[json.exception.type_error.305] cannot use operator[] with a string argument with array" thrown in the test body.
```

Reply:
Adding the `&` seems to fix it. 
E.g. `const NL::json &schema {m_info->schema()};`

Reply:
I don't understand why that would change anything.  Using dump() instead of get() makes sense, but I don't see how holding the reference rather than using a copy would change behavior.  I also don't understand why tests are failing for you but not on our platforms.  What compiler/OS version do you have?

Reply:
Ubuntu 18.04. With holding the reference, when I debug with Clion, I can see the string values. Otherwise I error out. 

```
jason@glassbox ~/w/e/build (master)> uname -a
Linux glassbox 4.15.0-48-generic #51-Ubuntu SMP Wed Apr 3 08:28:49 UTC 2019 x86_64 x86_64 x86_64 GNU/Linux

jason@glassbox ~/w/e/build (master)> cmake --version
cmake version 3.10.2

CMake suite maintained and supported by Kitware (kitware.com/cmake).
jason@glassbox ~/w/e/build (master)> gcc --version
gcc (Ubuntu 7.4.0-1ubuntu1~18.04) 7.4.0
Copyright (C) 2017 Free Software Foundation, Inc.
This is free software; see the source for copying conditions.  There is NO
warranty; not even for MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.
```